app/controllers/topics_controller.rb

```diff
@@ -538,6 +544,12 @@ def invite
       group_names: params[:group_names]
     )
 
+    unless pm_has_slots?(topic)
```

Comment:
This does not feel right cause you are allowed it appears to invite multiple groups, so you can cheat and invite 30 groups ? no?

Reply:
Yes that will allow you to cheat and invite 30 groups. Maybe we should have another site setting for max groups in a PM? Because making every group member take up a slot doesn't feel quite right to me.

Reply:
I think this is fine... just add `pm_has_slots?(topic, groups.length)` or something like that.

Reply:
the `groups` variable here represents groups that you want invited user to be a member of, when you invite by email. These groups won't be invited to the message.

Inviting a group to a message is handled via a separate controller method `invite_group`, and you can't invite groups in bulk, only one by one.

We do a check like this
`topic_allowed_users.count + topic_allowed_groups.count >= SiteSetting.max_allowed_message_recipients`
when you invite a group or user.